CASMNET-1534 - manual backport of CASMNET-1534-csm-1.2 to main (#1935)
Cray-HPE/docs-csm: operations/network/management_network/bican_support_matrix.md

| System Access to External/Site (LDAP, DNS) | Egress                     | Allowed                   | Allowed                | Not Allowed                  |

* Selection of user access for job control and data movement over the Shasta Management Network (CAN) or the High Speed Network (CHN) is made during system installation or upgrade.

* Creation of the Customer Management Network (CMN) during installation or upgrade is mandatory.

## Network overview

![tds can overview](img/tds_can_overview.png)

### Internal networks

* Node Management Network (NMN) \- Provides the internal control plane for systems management and jobs control.
* Hardware Management Network (HMN) \- Provides internal access to system baseboard management controllers (BMC/iLO) and other lower-level hardware access.

### External and edge networks

* Customer Management Network (CMN) \- Provides customer access from the site to the system for administrators.
* Customer Access Network (CAN) or Customer High Speed Network (CHN) provide:
  * Customer access from the site to the system for job control and jobs data movement.

>   * Access from the system to the site for network services like DNS, LDAP, etc.

Reviewer: ```suggestion
  * Access from the system to the site for network services like DNS, LDAP, and so on.
```

## Supported configurations

### Option A: CMN + CAN (Management Network only - Layer 2 separation)

![cmn plus can](img/cmn_plus_can.png)

### Option B: CMN + CHN (Administration over Management Network, User Access over High Speed Network)

![cmn plus chn](img/cmn_plus_chn.png)

> Note: During installation, the High Speed Network is not configured until relatively late in the process.

Reviewer: ```suggestion
Note: During installation, the High Speed Network (HSN) is not configured until relatively late in the process.
```

Installation generally requires site access for deployment artifacts, site DNS, etc.

> In order to achieve this, the CMN is used during the installation process for system traffic egress until the High Speed Network is available.

Reviewer: ```suggestion
In order to achieve this, the CMN is used during the installation process for system traffic egress until the HSN is available.
```

## Network capabilities

### Layer 2

* CMN, CAN, and CHN have broadcast boundaries at the edge between the system and the site.

### Layer 3

* Addressing
  * IPv4 supported (default)
  * IPv6 roadmap
* Routing
  * Static routes (default) exist on the edge router/switches.
  * Dynamic routing (OSPF) is possible at the edge.

## Network sizing and requirements

### CMN sizing and requirements

#### CMN IPv4 sizing and requirements

* Site routable
* Contiguous (CIDR block)
* Non-overlapping with internal networks (configurable during installation)
* Size estimate is the sum of:
  * Number of Non-Compute Nodes (NCNs) of type master, worker, or storage used by the Kubernetes cluster
  * Number of switches on the Management Network
  * Number of administrative API endpoints

>   * Several administrative addresses for switch interfaces and routing.

Reviewer: ```suggestion
  * Several administrative addresses for switch interfaces and routing
```

>   * A `/26` block is typically sufficient for systems less than approximately 4000 nodes.

Reviewer: ```suggestion
  * A `/26` block is typically sufficient for systems with less than approximately 4000 nodes
```

### CAN or CHN sizing and requirements

#### CAN or CHN IPv4 sizing and requirements

* Site routable
* Contiguous (CIDR block)
* Non-overlapping with internal networks (configurable during installation)
* Size estimate is the sum of:

>   * Number of application nodes requiring access from the site (User Access Node (UAN), login nodes, etc.)

Reviewer: ```suggestion
  * Number of application nodes requiring access from the site (User Access Node (UAN), login nodes, and more)
```

>   * Number of User Access Instances (UAI) in Kubernetes (if used).

Reviewer: ```suggestion
  * Number of User Access Instances (UAI) in Kubernetes (if used)
```

  * Number of API endpoints
  * Several administrative addresses for switch interfaces and routing

>   * NOTE: CAN or CHN sizing is largely dependent on customer-specific use cases and application node hardware.

Reviewer: ```suggestion
  * CAN or CHN sizing is largely dependent on customer-specific use cases and application node hardware
```
